pkg/controllers/jobset_controller.go

```diff
@@ -375,11 +377,15 @@ func (r *JobSetReconciler) suspendJobs(ctx context.Context, js *jobset.JobSet, a
 			}
 		}
 	}
-	setJobSetSuspended(js, updateStatusOpts)
+	setJobSetSuspendedCondition(js, updateStatusOpts)
 	return nil
 }
 
-func (r *JobSetReconciler) resumeJobsIfNecessary(ctx context.Context, js *jobset.JobSet, activeJobs []*batchv1.Job, replicatedJobStatuses []jobset.ReplicatedJobStatus, updateStatusOpts *statusUpdateOpts) error {
+// resumeJobsIfNecessary iterates through each replicatedJob, resuming any suspended jobs if the JobSet
+// is not suspended. Returns a boolean value indicating if the JobSet should be requeued for reconciliation.
+// This is so in-order startup policy can be respected, where after resuming one replicatedJob, we must
+// wait for it to become ready before resuming the next.
```

Comment:
Do we really need to requeue? wouldn't the status update of the child job trigger a requeue anyway?

Reply:
If there's multiple replicated jobs, and we are resuming one at a time (returning early after each replicatedJob is resumed so we can wait for it to be ready before resuming the next), then only the first status update to "in order startup policy in progress" will actually result in a status update API call, since the next replicated job we resume would attempt to set this same "in progress" status again, which would have no affect, since for duplicate conditions we make no update.

Reply:
The update to the status of the child jobs when they become ready will trigger a reconcile, i was not referring to the update to jobset status.

Reply:
Ah I see what you mea now, yes you're right. Updated the code to remove the requeue.